clientdb/batch_snapshot_test.go

```diff
@@ -112,44 +108,91 @@ var testSnapshot = &LocalBatchSnapshot{
 				NodeKey:     [33]byte{2, 2, 3},
 				NodeAddrs: []net.Addr{
 					&net.TCPAddr{
-						IP:   net.IP{0x12, 0x34, 0x56, 0x78},
+						IP: net.IP{
+							0x12, 0x34, 0x56, 0x78,
+						},
 						Port: 8080,
 					},
 				},
 
 				UnitsFilled: 10,
 			},
 		},
-	},
+	}
+
+	testSnapshot = &LocalBatchSnapshot{
+		Version: order.DefaultBatchVersion,
+		BatchID: testBatchID,
+		ClearingPrices: map[uint32]order.FixedRatePremium{
+			testDuration:     999,
+			testDuration * 2: 1234,
+		},
+		ExecutionFee:   *terms.NewLinearFeeSchedule(101, 900),
+		BatchTX:        testBatchTx,
+		BatchTxFeeRate: 123456,
+		Accounts:       testAccounts,
+		Orders:         testOrders,
+		MatchedOrders:  testMatchedOrders,
+	}
+
+	allSnapshots = []*LocalBatchSnapshot{
+		testSnapshot,
+		{
+			Version: order.DefaultBatchVersion,
+			BatchID: testBatchID,
+			ClearingPrices: map[uint32]order.FixedRatePremium{
+				order.LegacyLeaseDurationBucket: 999,
+			},
+			ExecutionFee:   *terms.NewLinearFeeSchedule(101, 900),
+			BatchTX:        testBatchTx,
+			BatchTxFeeRate: 123456,
+			Accounts:       testAccounts,
+			Orders:         testOrders,
+			MatchedOrders:  testMatchedOrders,
+		},
+	}
+)
+
+func newOrderKit(nonce order.Nonce, duration uint32) *order.Kit {
+	kit := order.NewKit(nonce)
+	kit.LeaseDuration = duration
+	return kit
 }
 
 // TestSerializeLocalBatchSnapshot checks that (de)serialization of local batch
 // snapshots works as expected.
 func TestSerializeLocalBatchSnapshot(t *testing.T) {
-	pre := testSnapshot
-	buf := bytes.Buffer{}
-	if err := serializeLocalBatchSnapshot(&buf, pre); err != nil {
-		t.Fatal(err)
-	}
+	for idx, testSnapshot := range allSnapshots {
+		pre := testSnapshot
+		t.Run(fmt.Sprintf("snapshot-%d", idx), func(t *testing.T) {
+			t.Parallel()
 
-	post, err := deserializeLocalBatchSnapshot(&buf)
-	if err != nil {
-		t.Fatal(err)
-	}
+			buf := bytes.Buffer{}
+			if err := serializeLocalBatchSnapshot(&buf, pre); err != nil {
+				t.Fatal(err)
+			}
 
-	if !reflect.DeepEqual(pre, post) {
-		for _, a := range pre.Accounts {
-			a.TraderKey.PubKey.Curve = nil
-			a.AuctioneerKey.Curve = nil
-			a.BatchKey.Curve = nil
-		}
-		for _, a := range post.Accounts {
-			a.TraderKey.PubKey.Curve = nil
-			a.AuctioneerKey.Curve = nil
-			a.BatchKey.Curve = nil
-		}
+			post, err := deserializeLocalBatchSnapshot(&buf)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(pre, post) {
```

Comment:
Nit: Is `require.Equal` any better here?

Reply:
Yes, the diff seems to be much cleaner :+1: